src/snyk/snykOss/views/suggestion/suggestionViewProvider.ts

```diff
@@ -0,0 +1,201 @@
+      !this.panel?.viewColumn ||
+      this.panel.viewColumn !== vscode.ViewColumn.Two
+    ) {
+      // workaround for: https://github.com/microsoft/vscode/issues/71608
+      // when resolved, we can set showPanel back to sync execution.
+      await vscode.commands.executeCommand('workbench.action.focusSecondEditorGroup');
+    }
+
+    if (this.panel) {
+      this.panel.reveal(vscode.ViewColumn.Two, true);
+    } else {
+      this.panel = vscode.window.createWebviewPanel(
+        SNYK_VIEW_SUGGESTION_OSS,
+        'Snyk OSS Vulnerability',
+        {
+          viewColumn: vscode.ViewColumn.Two,
+          preserveFocus: true,
+        },
+        {
+          enableScripts: true,
+          localResourceRoots: [this.context.getExtensionUri()],
+        },
+      );
+    }
+
+    this.panel.webview.html = this.getHtmlForWebview(this.panel.webview);
+
+    void this.panel.webview.postMessage({ type: 'set', args: vulnerability });
+
+    this.panel.onDidDispose(this.onPanelDispose.bind(this));
+    this.panel.webview.onDidReceiveMessage((data: FeaturesViewEventMessage) => {
+      switch (data.type) {
+        case SuggestionViewEventMessageType.OpenBrowser:
+          void vscode.commands.executeCommand(SNYK_OPEN_BROWSER_COMMAND, data.value);
+          break;
+        default:
+          break;
+      }
+    });
+    this.panel.onDidChangeViewState(this.checkVisibility.bind(this));
+  }
+
+  restorePanel(panel: vscode.WebviewPanel): void {
+    if (this.panel) this.panel.dispose();
+    this.panel = panel;
+  }
+
+  private disposePanel() {
+    if (this.panel) this.panel.dispose();
+  }
+
+  private onPanelDispose() {
+    this.panel = undefined;
+  }
+
+  private checkVisibility(_e: vscode.WebviewPanelOnDidChangeViewStateEvent): void {
+    if (this.panel && this.panel.visible) {
+      void this.panel.webview.postMessage({ type: 'get' });
+    }
+  }
+
+  private getWebViewUri(...pathSegments: string[]) {
+    return this.panel?.webview.asWebviewUri(vscode.Uri.joinPath(this.context.getExtensionUri(), ...pathSegments));
+  }
+
+  private getHtmlForWebview(webview: vscode.Webview) {
+    const images: Record<string, string> = [
+      ['icon-code', 'svg'],
+      ['dark-critical-severity', 'svg'],
+      ['dark-high-severity', 'svg'],
+      ['dark-medium-severity', 'svg'],
+      ['dark-low-severity', 'svg'],
+    ].reduce((accumulator: Record<string, string>, [name, ext]) => {
+      const uri = this.getWebViewUri('images', `${name}.${ext}`); // todo move to media folder
+      if (!uri) throw new Error('Image missing.');
+      accumulator[name] = uri.toString();
+      return accumulator;
+    }, {});
+
+    const scriptUri = this.getWebViewUri('out', 'snyk', 'snykOss', 'views', 'suggestion', 'suggestionViewScript.js');
+    const styleUri = this.getWebViewUri('media', 'views', 'oss', 'suggestion', 'suggestion.css');
+
+    const nonce = getNonce();
+
+    return `<!DOCTYPE html>
```

Comment:
is there a way we could have a standalone html file for this script to use as a template, or is it an overkill?

Reply:
We'd have to surface it from the file system that might give some penalty, see the [comment above](https://github.com/snyk/vscode-extension/pull/53/files/af5b4c30db22946bb8b672150fb8b14903f0ea97#r707364894). I didn't want to introduce major changes around that as it'd require change around Snyk Code for consistency which is not a part of this ticket.